In [14]:
from scipy.stats import chi2_contingency
import numpy as np

row1 = [91,90,51]
row2 = [150,200,155]
row3 = [109,198,172]
data = [row1, row2, row3]
print(chi2_contingency(data))

chi2, p_value, dfreedom, expected = chi2_contingency(data)

(25.085973274234959, 4.8346447416999636e-05, 4, array([[  66.77631579,   93.10526316,   72.11842105],
       [ 145.35361842,  202.66447368,  156.98190789],
       [ 137.87006579,  192.23026316,  148.89967105]]))


### R output

    Pearson's Chi-squared test

    data:  data.table 
    X-squared = 25.086, df = 4, p-value = 4.835e-05

In [13]:
print(expected.sum().sum())
print(np.array(data).sum().sum())

1216.0
1216
